### ISSUE TITLE
Stop modification of br tag

### ISSUE THREAD

**Issue by milne-dev** [apostrophecms/sanitize-html#51]
When I pass `<br>` through the `sanitizeHtml()` is there any way to keep it from changing to `<br />` ?

Or for other tags that may be similiar..

Thanks!

**Comment by mathieumg** [comment]
`<br>` is not valid whereas `<br />` is. What is your use case?

**Comment by simison** [comment]
<br> is perfectly valid HTML, <br/> valid XHTML. ;-)

http://w3c.github.io/html-reference/syntax.html#void-element
>Optionally, a "/" character, which may be present only if the element is a void element.

**Comment by boutell** [comment]
As long as htmlparser2 doesn't offer a way to sniff out whether the original tag had a / or not, sanitize-html will continue to use its internal list of void elements to decide whether to use /.

But even if it did, I'm not sure I would change this. `<br>` may be acceptable in HTML, but `<br />` is always 100% valid as both HTML and XHTML. It's just cleaner and it does not change the meaning of the code in any way.

**Comment by Michael77** [comment]
For large deployments that store a lot of user generated html, it is favorable to keep the data size as low as possible.

**Comment by omerts** [comment]
@boutell I am having a similar issue with an image tag. It is totally breaking my functionality since I am working with innerHTML attribute which return the html like it is in the browser's DOM. Chrome strips out self closing tags, which causes html that I pass through html-sanitize to be inconsistent with my innerHTML. Consistency is critical for me, since I am changing the html by selection indexes, and if the sanitize html changes it, indexes are offset.

**Comment by matiishyn** [comment]
I have same issue here
Please, Stop modification of br tag

**Comment by dman777** [comment]
I am hitting this issue with quill. I don't think quill likes <br />

**Comment by gastonyte** [comment]
Hello, Thank you for this library.

I'm also a user of Quill and I'd find it helpful to have an option to disable the addition of a slash for `<br>` tags.

I wouldn't say "it's just cleaner", but "this is one of the possible correct solutions".

For those interested in a solution to this particular problem, I've made it possible in my [fork](https://github.com/gastonyte/sanitize-html/tree/c54eeb1624fcb41bdb49f57ca2530ae9d8de07c3):

To achieve this, define the `selfClosing` option as an object. The keys of this object should be the tag names, and the value should be a truthy or falsy value. If the value of a specific tag has the property `voidTag` set to true, the library will not insert the space and the slash.

**Comment by boutell** [comment]
These are really two separate features:

1. "Preserve it the way it was" ("stop modification," as the title says)

We can't easily do this because of the way the upstream module gives us the information.

2. "An option to not use the /"

Sure, that would be fine as long as the default doesn't change unexpectedly. @gastonyte would you like to create a PR for your feature?

**Comment by gastonyte** [comment]
Hi @boutell, I didn't know if this feature was wanted or not, so I just made the modification in a fork to be able to use it in my app, but if it's something you would like to see integrated into your library, it is with pleasure that I will create the PR =)

**Comment by boutell** [comment]
Yes, supporting this via a flag sounds great! My objection was to the original request in which the original presence or absence of a `/` would be preserved, simply because I don't think we have access to that information.

**Comment by milne-dev** [comment]
I'll comment to keep this open since someone created a PR
